File: JS Programming Basics/05. While Loop - Exercise/vacation.js
function solve (input){
    let neededMoney = Number(input[0]);
    let money = Number(input[1]);
    let command = input[2];

    let currentMoney = Number(input[3]);
    let spendingDays =0;
    let totalDays = 0;

    for(let i =4; i<=input.length; i++){
        if(command === "spend"){
            money -= currentMoney;
                if(money<0){
                    money=0;
                }
            spendingDays++;
            totalDays++;
        } else if (command === "save"){
            money += currentMoney;
            totalDays++;
            spendingDays=0;
        }

        if(neededMoney === money){
            break;
        }

        if(spendingDays>=5){
            console.log(`You can't save the money.`);
            console.log(totalDays);
            break;
        }

        
        command = input[i];
        i++;
        currentMoney=Number(input[i]);

    }

    if(spendingDays!==5){
        console.log(`You saved the money for ${totalDays} days.`);
    }
}

solve(["250",
"150",
"spend",
"50",
"spend",
"50",
"save",
"100",
"save",
"100"])

File: Programming Basics with Java/05 While Loop - Exercise/src/Vacation.java
import java.util.Scanner;

public class Vacation {
    public static void main(String[] args) {
        Scanner scan = new Scanner (System.in);

        double moneyNeeded = Double.parseDouble(scan.nextLine());
        double money = Double.parseDouble(scan.nextLine());
        int daysCount = 0;
        int spendingDays = 0;

        while(money < moneyNeeded && spendingDays < 5){
            String action = scan.nextLine();
            double currentSum = Double.parseDouble(scan.nextLine());
            daysCount++;

            if(action.equals("spend")){
                spendingDays ++;
                money -= currentSum;

                if(money < 0){
                    money = 0;
                }

            } else if(action.equals("save")){
                spendingDays = 0;
                money += currentSum;
            }
        }

        if(spendingDays == 5){
            System.out.println("You can't save the money.");
            System.out.println(daysCount);
        } else {
            System.out.printf("You saved the money for %d days.", daysCount);
        }

    }
}